commit.go

```diff
@@ -221,32 +221,61 @@ func (b *Builder) Commit(ctx context.Context, dest types.ImageReference, options
 			logrus.Warnf("don't know how to add tags to images stored in %q transport", dest.Transport().Name())
 		}
 	}
-
-	img, err := is.Transport.GetStoreImage(b.store, dest)
-	if err != nil && err != storage.ErrImageUnknown {
-		return imgID, nil, "", errors.Wrapf(err, "error locating image %q in local storage", transports.ImageName(dest))
-	}
+	// Calculate the image ID, now that we have a real image with a
+	// configuration blob that's in its final format.
+	img, err := dest.NewImage(ctx, systemContext)
```

Comment:
If `dest` is not local storage (and, strictly speaking, even if it were), nothing guarantees that `dest` points to exactly the just written image (e.g. if `dest` is a tag reference, the tag can be reassigned at any time).

That’s, to an extent, a weakness in the `ImageReference` design — but transports that write into mutable shared storage just don’t have stable references to immutable content (even digest references don’t work, because the image may be removed in the meantime), so any write+read is never guaranteed to return the originally written data in principle.

Reply:
Aargh, that's right, the tag can move in the interim.